lldb/source/Interpreter/ScriptInterpreter.cpp

```diff
@@ -104,6 +104,19 @@ ScriptInterpreter::GetStatusFromSBError(const lldb::SBError &error) const {
   return Status();
 }
 
+Event *
+ScriptInterpreter::GetOpaqueTypeFromSBEvent(const lldb::SBEvent &event) const {
+  return event.m_opaque_ptr;
+}
+
+Stream *ScriptInterpreter::GetOpaqueTypeFromSBStream(
+    const lldb::SBStream &stream) const {
+  if (stream.m_opaque_up)
+    return const_cast<lldb::SBStream &>(stream).m_opaque_up.release();
```

Comment:
This feels a little sketchy to me... `GetOpaqueTypeFromSBStream` is taking ownership of the underlying pointer and returning it to the caller. The `SBStream` that gets passed in is no longer valid. Is this intentional? That would mean this function -- and only this function -- has different semantics from the other `GetOpaqueTypeFromSBClass` functions.

Reply:
`lldb::SBStream` have different semantics than other classes, as it uses a unique_ptr to point to its opaque type and  also `lldb_private::Stream` is an abstract type so the only way to get the underlying object is to do this. Let me know if you can think of a better way of doing it.

Reply:
I'm having a hard time wrapping my head around the SBStream portion of your patch (everything else makes sense). So we pass in a `const SBStream &` to `GetOpaqueTypeFromSBStream` which that takes ownership of the underlying `Stream *`  and returns it. What happens to the original SBStream argument? Can we guarantee that nothing ever tries to use it again afterwards?
It looks like the intent in `ReverseTransform` is to take ownership of the original user-provided stream and write its contents to some other stream `original_arg` after performing some kind of copy on it. Instead of taking ownership of the underlying stream (and invalidating its argument), why not just use `.get()` instead of `.release()`? The `SBStream` passed in here won't be invalidated and `original_arg` in `ReverseTransform` can still get the data out of the Python `SBStream` object.

What am I missing?

Reply:
Makes sense. Updated the PR.